MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/ScalebarPluginActivity.java

```diff
@@ -0,0 +1,169 @@
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+    setContentView(R.layout.activity_scalebar_plugin);
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+    mapboxMap.setStyle(listOfStyles[index], new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull final Style style) {
+
+        initStyling();
+
+        // Create a new ScaleBarPlugin object
+        scaleBarPlugin = new ScaleBarPlugin(mapView, mapboxMap);
+
+        scaleBarPlugin.create(listOfScalebarStyleVariations[index]);
+
+        findViewById(R.id.switch_scalebar_style_fab).setOnClickListener(new View.OnClickListener() {
+          @Override
+          public void onClick(View view) {
+
+            if (index == listOfScalebarStyleVariations.length - 1) {
+              index = 0;
+            }
+
+            mapboxMap.setStyle(listOfStyles[index], new Style.OnStyleLoaded() {
+              @Override
+              public void onStyleLoaded(@NonNull Style style) {
+
+                scaleBarPlugin.create(listOfScalebarStyleVariations[index]);
+
+              }
+            });
+
+            index++;
+
+          }
+        });
+
+        Toast.makeText(ScalebarPluginActivity.this, getString(R.string.zoom_map_fab_instruction),
+          Toast.LENGTH_LONG).show();
+      }
+    });
+  }
+
+  private void initStyling() {
+    listOfScalebarStyleVariations = new ScaleBarOptions[] {
+
+      // Using the plugin's default styling to start
+      new ScaleBarOptions(this),
+
+      // Random styling option #2
+      new ScaleBarOptions(this)
+        .setTextColor(R.color.mapboxRed)
+        .setTextSize(40f)
+        .setBarHeight(15f)
+        .setBorderWidth(5f)
+        .setMetricUnit(true)
+        .setRefreshInterval(15)
+        .setMarginTop(30f)
+        .setMarginLeft(16f)
+        .setTextBarMargin(15f),
+
+      // Random styling option #3
+      new ScaleBarOptions(this)
+        .setTextColor(R.color.mapbox_blue)
+        .setTextSize(60f)
+        .setBarHeight(15f)
+        .setBorderWidth(5f)
+        .setMetricUnit(true)
+        .setRefreshInterval(15)
+        .setMarginTop(30f)
+        .setMarginLeft(30f)
+        .setTextBarMargin(25f),
+
+      // Random styling option #4
+      new ScaleBarOptions(this)
+        .setTextColor(R.color.mapboxYellow)
+        .setTextSize(30f)
+        .setBarHeight(15f)
+        .setBorderWidth(5f)
+        .setMetricUnit(false)
+        .setRefreshInterval(15)
+        .setMarginTop(30f)
+        .setMarginLeft(30f)
+        .setTextBarMargin(25f),
```

Comment:
We can add a new option with `@DimenRes` parameters.

Reply:
You're mean something like

```java
private ScaleBarOptions createScaleBarOptions(
      @DimenRes int textSize,
      @DimenRes int barHeight,
      @DimenRes int borderWidth,
      @DimenRes int refreshInterval,
      @DimenRes int marginTop,
      @DimenRes int marginLeft,
      @DimenRes int textBarMargin) {

    // Random styling option #4
    return new ScaleBarOptions(this)
        .setTextColor(R.color.mapboxYellow)
        .setTextSize(textSize)
        .setBarHeight(barHeight)
        .setBorderWidth(borderWidth)
        .setMetricUnit(false)
        .setRefreshInterval(refreshInterval)
        .setMarginTop(marginTop)
        .setMarginLeft(marginLeft)
        .setTextBarMargin(textBarMargin);
  }
```

and then use `createScaleBarOptions()` ?

Reply:
Something like this, define values in `dimens.xml`
```
        scaleBarOptions
                .setTextColor(R.color.mapboxRed)
                .setTextSize(resources.getDimension(R.dimen.fab_margin))
                .setBarHeight(resources.getDimension(R.dimen.fab_margin))
                .setBorderWidth(resources.getDimension(R.dimen.fab_margin))
                .setMetricUnit(true)
                .setRefreshInterval(15)
                .setMarginTop(resources.getDimension(R.dimen.fab_margin))
                .setMarginLeft(resources.getDimension(R.dimen.fab_margin))
                .setTextBarMargin(resources.getDimension(R.dimen.fab_margin))
```

Reply:
but then the values would be set to the same number, no? If you look at the GIF at the top, I cycle through various `scaleBarOptions` sets 

https://github.com/mapbox/mapbox-android-demo/blob/bf0f89cce507a108325df8ff0e2b88936364ca67/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/ScalebarPluginActivity.java#L91-L124

Reply:
What I pasted here is only an example, we should define these values in `dimens.xml`

Reply:
oh. I think it would be better for the values to just be visible in the example activity itself. That way they're visible when viewed on Github and when the example is on https://docs.mapbox.com/android/plugins/examples/. That way, folks don't have to go hunting in the XML file for the exact value.

Reply:
Got it, no need to define dimens.

Reply:
Can I get a ✅ from you unless there are any other needed changes?